backend/src/tests/tests.test.js

```diff
@@ -0,0 +1,191 @@
+const typeDefs = require('../typeDefs');
+const resolvers = require('../resolvers');
+
+let usersApi = new UsersAPI();
+let postsApi = new PostsAPI();
+
+const server = new ApolloServer({
+    typeDefs, 
+    resolvers, 
+    dataSources: () => {
+        return {
+          usersApi: usersApi,
+          postsApi: postsApi
+        }
+    }
+});
+
+const { query, mutate } = createTestClient(server);
+
+describe("query", () => {
+
+    it("posts with nested user object", async () => {
+        const GET_POSTS = gql`
+            query {
+                posts {
+                    title 
+                    author {
+                        name
+                    }
+                }
+            }
+        `;
+
+        const {
+            data: { posts }
+        } = await query({ query: GET_POSTS });
+
+        expect(posts).toHaveLength(2);
+        expect(posts).toEqual([
+            {
+                title: 'The Thing',
+                author: {
+                    name: "Peter"
+                },
+            },
+            {
+                title: 'The Nothing',
+                author: {
+                    name: "Peter"
+                },
+            }
+        ]);
+    });
+
+    it("users with nested post object", async () => {
+        const GET_USERS = gql`
+            query {
+                users {
+                    name 
+                    posts {
+                        title
+                        votes
+                    }
+                }
+            }
+        `;
+
+        const {
+            data: { users }
+        } = await query({ query: GET_USERS });
+
+
+        expect(users).toHaveLength(2);
+        expect(users).toEqual([
+            {
+                name: "Peter",
+                posts: [
+                    {
+                        title: "The Thing",
+                        votes: 0
+                    },
+                    {
+                        title: "The Nothing",
+                        votes: 0
+                    }
+                ]
+            }, 
+            {
+                name: "Max",
+                posts: []
+            }
+        ]);
+    });
+  
+});
+
+describe("write(post: $postInput)", () => {
+
+    let WRITE_POST;
+
+    beforeEach(() => {
+        WRITE_POST = gql`
+            mutation WritePost($post: PostInput!) {
+                write(post: $post) {
+                    title, 
+                    author {
+                        name
+                    }
+                }
+            }
+        `;
+    })
+
+    it("creates new post", async () => {
+        const {
+            data: { write }
+        } = await mutate({ mutation: WRITE_POST, variables: { post: { title: "Title1", author: { name: "Max" } } } });
+
+        expect(write).toEqual({title: "Title1", author: {name: "Max"}});
+    });
+
+    it("throws error if author creates post with already existent title", async () => {    
```

Comment:
Why didn't you implement the schema as described in the exercise description?

In the exercise description:
```gql
  type Post {
    id: ID!
    title: String!
    votes: Int!
    author: User!
  }

```

Yours:
```gql
  type Post {
    title: ID!
    votes: Int!
    author: User!
  }
```

Reply:
we implemented it in this way, because our Frontend expects the title to be unique, aka ID